pkg/operator/csi/csidefaultstorageclasscontroller/csi_default_storageclass_controller.go

```diff
@@ -0,0 +1,153 @@
+// It produces following Conditions:
+// DefaultStorageClassControllerDegraded - failed to apply StorageClass provided
+type CSIDefaultStorageClassController struct {
+	name               string
+	assetFunc          resourceapply.AssetFunc
+	file               string
+	kubeClient         kubernetes.Interface
+	storageClassLister v1.StorageClassLister
+	operatorClient     v1helpers.OperatorClient
+	eventRecorder      events.Recorder
+}
+
+func NewCSIDefaultStorageClassController(
+	name string,
+	assetFunc resourceapply.AssetFunc,
+	file string,
+	kubeClient kubernetes.Interface,
+	InformerFactory informers.SharedInformerFactory,
+	operatorClient v1helpers.OperatorClient,
+	eventRecorder events.Recorder) factory.Controller {
+	c := &CSIDefaultStorageClassController{
+		name:               name,
+		assetFunc:          assetFunc,
+		file:               file,
+		kubeClient:         kubeClient,
+		storageClassLister: InformerFactory.Storage().V1().StorageClasses().Lister(),
+		operatorClient:     operatorClient,
+		eventRecorder:      eventRecorder,
+	}
+
+	return factory.New().WithSync(
+		c.Sync,
+	).ResyncEvery(
+		time.Minute,
+	).WithSyncDegradedOnError(
+		operatorClient,
+	).WithInformers(
+		operatorClient.Informer(),
+		InformerFactory.Storage().V1().StorageClasses().Informer(),
+	).ToController(
+		"DefaultStorageClassController",
+		eventRecorder,
+	)
+}
+
+func (c *CSIDefaultStorageClassController) Sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	klog.V(4).Infof("DefaultStorageClassController sync started")
+	defer klog.V(4).Infof("DefaultStorageClassController sync finished")
+
+	opSpec, _, _, err := c.operatorClient.GetOperatorState()
+	if err != nil {
+		return err
+	}
+	if opSpec.ManagementState != operatorapi.Managed {
+		return nil
+	}
+
+	syncErr := c.syncStorageClass(ctx)
+	if syncErr != nil {
+		degradedCnd := operatorapi.OperatorCondition{
+			Type:    conditionsPrefix + operatorapi.OperatorStatusTypeDegraded,
+			Status:  operatorapi.ConditionTrue,
+			Reason:  "SyncError",
+			Message: syncErr.Error(),
+		}
+
+		_, _, updateErr := v1helpers.UpdateStatus(ctx, c.operatorClient,
+			v1helpers.UpdateConditionFn(degradedCnd),
+		)
+		return updateErr
+	}
+
+	if _, _, updateErr := v1helpers.UpdateStatus(ctx, c.operatorClient,
+		removeConditionFn(conditionsPrefix+operatorapi.OperatorStatusTypeDegraded),
+	); updateErr != nil {
+		return errutil.NewAggregate([]error{syncErr, updateErr})
+	}
+
+	return syncErr
+}
+
+func (c *CSIDefaultStorageClassController) syncStorageClass(ctx context.Context) error {
+	expectedScBytes, err := c.assetFunc(c.file)
+	if err != nil {
+		return err
+	}
+
+	expectedSC := resourceread.ReadStorageClassV1OrDie(expectedScBytes)
+
+	existingSCs, err := c.storageClassLister.List(labels.Everything())
+	if err != nil {
+		klog.V(2).Infof("could not list StorageClass objects")
+		return err
+	}
+
+	defaultScExists := false
+	defaultScCount := 0
```

Comment:
`defaultScExists` seems to be useless, as you can always check `defaultScCount > 0`

Reply:
Fixed.